src/cpu/aarch64/matmul/acl_lowp_matmul_sq.hpp

```diff
@@ -0,0 +1,326 @@
+            using smask_t = primitive_attr_t::skip_mask_t;
+            VDISPATCH_MATMUL(
+                    attr()->has_default_values(smask_t::scales_runtime
+                            | smask_t::zero_points_runtime | smask_t::post_ops),
+                    "only scale, zero point and post-ops attrs supported");
+
+            VDISPATCH_MATMUL(attr()->scales_.get(DNNL_ARG_SRC).mask_ == 0
+                            && attr()->zero_points_.get(DNNL_ARG_SRC) == 0
+                            && attr()->scales_.get(DNNL_ARG_WEIGHTS).mask_ == 0
+                            && attr()->zero_points_.get(DNNL_ARG_WEIGHTS) == 0
+                            && attr()->scales_.get(DNNL_ARG_DST).mask_ == 0
+                            && attr()->zero_points_.get(DNNL_ARG_DST) == 0,
+                    "common scales and zero points only");
+
+            VDISPATCH_MATMUL(!has_runtime_dims_or_strides(),
+                    VERBOSE_RUNTIMEDIM_UNSUPPORTED);
+
+            const memory_desc_wrapper src_d(src_md_);
+            const memory_desc_wrapper wei_d(weights_md_);
+            const memory_desc_wrapper bia_d(bias_md_);
+            const memory_desc_wrapper dst_d(dst_md_);
+
+            using namespace data_type;
+            VDISPATCH_MATMUL(utils::one_of(src_d.data_type(), s8, u8)
+                                    && wei_d.data_type() == s8
+                                    && src_d.data_type() == s8
+                            ? dst_d.data_type() == s8
+                            : dst_d.data_type() == u8
+                                    && utils::one_of(
+                                            bia_d.data_type(), f32, undef),
+                    VERBOSE_UNSUPPORTED_DT_CFG);
+
+            VDISPATCH_MATMUL(src_d.matches_tag(format_tag::ab)
+                            && wei_d.matches_tag(format_tag::ab)
+                            && dst_d.matches_tag(format_tag::ab),
+                    VERBOSE_UNSUPPORTED_TAG);
+
+            VDISPATCH_MATMUL_SC(
+                    memory_desc_init_by_tag(bias_md_, bias_md_.ndims,
+                            bias_md_.dims, bias_md_.data_type, format_tag::ab),
+                    VERBOSE_UNSUPPORTED_BIAS_CFG);
+
+            // We set the QuantizationInfo to be dynamic because it is re-set in run()
+            almc_.src_tensor_info = arm_compute::TensorInfo(
+                    arm_compute::TensorShape(K(), M()), 1,
+                    acl_utils::get_acl_data_t(src_d.data_type(), true),
+                    arm_compute::QuantizationInfo(1.0, 0, true));
+            almc_.src_tensor_info.set_are_values_constant(false);
+
+            almc_.wei_tensor_info = arm_compute::TensorInfo(
+                    arm_compute::TensorShape(N(), K()), 1,
+                    acl_utils::get_acl_data_t(wei_d.data_type(), true),
+                    arm_compute::QuantizationInfo(1.0, 0, true));
+            almc_.wei_tensor_info.set_are_values_constant(true);
```

Comment:
Hi @fadara01 , 
What does constant mean here? Is it about constant values within the tensor or just that the shape is constant?
If it is about constant values in weights, it does not seem that primitive creation takes the weights, does it mean the primitive state changes upon first execution?